tools/plop-templates/component-index.hbs

```diff
@@ -1,11 +1,13 @@
 import * as React from 'react';
 import * as PropTypes from 'prop-types';
+import type {BoxProps} from '@twilio-paste/box';
 
-export interface {{pascalCase component-name}}Props {
+export interface {{pascalCase component-name}}Props extends Omit<React.ComponentPropsWithRef<'div'>, 'children'> {
```

Comment:
We noticed that we're not typing our components super specifically. We started using `extends Partial<Omit<HTMLDivElement, 'children'>>` in response. Recently, we used` extends Omit<React.ComponentPropsWithRef<'div'>, 'children'>` in the ButtonGroup package.

I would really like to get some clarity on:
- whether we even want to be this specific in our props(@TheSisb pointed out potential usability issues)
- which of these methods makes more sense for our system, if any
- which to include here in the template we're using for future components

Reply:
For consumers to make accessible applications, we have to extend the HTML element props that we expose to the consumer via the ref and where we spread our props onto. An Input has to reflect an input, a Select a select, etc. Our components really need to act like HTML elements, because underneath, that's all they are.

I really enjoyed the explanations in [this link](https://react-typescript-cheatsheet.netlify.app/docs/advanced/patterns_by_usecase/#wrappingmirroring-a-html-element) that @jb-twilio sent me on the subject, especially the "Why not ComponentProps or IntrinsicElements or [Element]HTMLAttributes or HTMLProps or HTMLAttributes" callout disclosure. Super good read.

Reply:
That's a good idea @jb-twilio - can you make a ticket for it pls?